src/components/trade/tradeWidget/useBuySell.ts

```diff
@@ -173,7 +173,11 @@ export const useBuySell = ({
 
   useEffect(() => {
     if (byTargetQuery.data) {
-      if (new Decimal(targetInput).gt(new Decimal(liquidityQuery.data || 0))) {
+      if (
```

Comment:
It's not super readable, could you create variables with good names so it's easier to understand what is this condition about

Reply:
Made it clearer

Reply:
we can consider creating a utility function, something in the lines of (coding from memory)
```ts

function tryDecimal(value: number|string|Decimal) : Decimal | NaN {
  try {
    return new Decimal(value);
  } catch {
    return NaN;
  }
}

```

Reply:
and of course use it only when the input is questionable

Reply:
Updated the implementation